hibernate-core/src/main/java/org/hibernate/dialect/SQLServerSqlAstTranslator.java

```diff
@@ -293,7 +297,12 @@ protected void renderFetchPlusOffsetExpression(
 			Expression fetchClauseExpression,
 			Expression offsetClauseExpression,
 			int offset) {
-		renderFetchPlusOffsetExpressionAsSingleParameter( fetchClauseExpression, offsetClauseExpression, offset );
+		if ( getCurrentQueryPart().isRoot() ) {
+			renderFetchPlusOffsetExpressionAsSingleParameter( fetchClauseExpression, offsetClauseExpression, offset );
+		}
+		else {
+			super.renderFetchPlusOffsetExpression( fetchClauseExpression, offsetClauseExpression, offset );
+		}
```

Comment:
I'm not quite sure what the purpose of this change is. Can you explain it? I believe SQL Server is picky about requiring a simple expression literal/parameter to appear in the top or fetch clause, so this code creates a single parameter instead of an addition expression `offset + limit`, but why would we only need that for the root?